### PR TITLE
Index permissions in document

### PR DESCRIPTION
**Pull request by wangch079**
Related discussion: https://elastic.slack.com/archives/C030MMRDFL3/p1649142679941249

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Related Pull Requests

* https://github.com/elastic/ent-search/pull/6421

### CODE REVIEW

File: lib/connectors_sdk/share_point/http_call_wrapper.rb [hunk 1]
```diff
@@ -23,10 +23,10 @@ def extractor(params)
         # XXX can we cache that class across calls?
         ConnectorsSdk::SharePoint::Extractor.new(
           content_source_id: BSON::ObjectId.new,
-          service_type: 'sharepoint_online',
+          service_type: SERVICE_TYPE,
```
Comment (wangch079) [inline lib/connectors_sdk/share_point/http_call_wrapper.rb:26]: `SERVICE_TYPE` is `sharepoint`, I don't think it make any more sense to use `sharepoint_online` here.

File: spec/connectors_sdk/sharepoint/http_call_wrapper_spec.rb [hunk 2]
```diff
@@ -157,20 +157,20 @@ def mock_endpoint(path, data)
   end
 
   context '.source_status' do
-    let(:access_token) { 'access token' }
+    let(:params) { { :access_token => 'access_token' } }
```
Comment (tarekziade) [inline spec/connectors_sdk/sharepoint/http_call_wrapper_spec.rb:160]: maybe we could add `:index_permissions` here and verify that `ConnectorsSdk::Office365::Config` gets it

File: spec/connectors_sdk/sharepoint/http_call_wrapper_spec.rb [hunk 3]
```diff
@@ -24,7 +24,10 @@
   end
 
   let(:params) do
-    { 'access_token' => 'something' }
+    {
+      'access_token' => 'something',
+      'index_permissions' => false
```
Comment (tarekziade) [inline spec/connectors_sdk/sharepoint/http_call_wrapper_spec.rb:29]: `false` is the default value in your code. can we switch this one to `true` and then assert that `ConnectorsSdk::Office365::Config.index_permissions` is `true` ?

**Review by tarekziade — APPROVED** [review]
Thanks! Is the `Gemfile.lock` changes intended?

**Comment by wangch079** [comment]
> Thanks! Is the `Gemfile.lock` changes intended?

When I ran `make test` locally, it was changed. Do you see the same in your local branch?

**Comment by tarekziade** [comment]
yeah I think it depends on the Ruby interpreter that is being used. I suspect some folks use JRuby through their IDE (like when @seanstory  committed this one https://github.com/elastic/connectors/commit/24f6a5cf4d4a61ba4f52dcf96125d023040d4dbd)

I think it's fine to push it like this, we should try to make sure we all use the same ruby when running the install

**Review by mchernyavskaya — APPROVED** [review]
LGTM

**Comment by tarekziade** [comment]
> > I suspect some folks use JRuby through their IDE (like when @seanstory committed this one [24f6a5c](https://github.com/elastic/connectors/commit/24f6a5cf4d4a61ba4f52dcf96125d023040d4dbd))
> > I think it's fine to push it like this, we should try to make sure we all use the same ruby when running the install
> 
> Yeah, I found at some point that I had an `RBENV_VERSION` env var set, and it was overrriding the `.ruby_version` at the root of the repo. :/ Rbenv _should_ take care of this for us.

To be honest, I have seen this many times in `ent-search` too. I wonder how we can make this more robust.

**Comment by seanstory** [comment]
@wangch079 I don't think this was backported to the 8.2 branch. This is intended for 8.2, per the label, is that correct?

**Comment by wangch079** [comment]
@seanstory , yes it's for 8.2. I have created the backport PR: https://github.com/elastic/connectors/pull/71

**Pull request merged**